tools/rewriter/DetermineAbi.cpp

```diff
@@ -66,13 +73,16 @@ static std::vector<CAbiArgKind> classifyDirectType(const clang::Type &type,
           pending_kind.reset();
         }
 
-        // Update pending_kind based on ABI rules
+        // Update pending_kind based on ABI rules.
+        // We're passing in a scalar so we expect we will only get one element.
```

Comment:
It isn't actually always the case that we pass a scalar type here. A struct like `struct s16 { struct { int a; int b; } inner; };` will also take this path for its first field but the recursive call does return a vector of a single eightbyte classification here because the inner struct fits in a single register.

Reply:
Fixed!